src/Metadata/Property/Factory/SerializerPropertyMetadataFactory.php

```diff
@@ -124,12 +126,16 @@ private function transformLinkStatus(PropertyMetadata $propertyMetadata, array $
             $relatedClass = $this->resourceClassResolver->getResourceClass(null, $relatedClass);
         }
 
-        $relatedGroups = $this->getClassSerializerGroups($relatedClass);
+        $outputClass = $this->getOutputClass($relatedClass) ?? $relatedClass;
+        $relatedGroups = $this->getClassSerializerGroups($outputClass);
 
         if (null === $propertyMetadata->isReadableLink()) {
             $propertyMetadata = $propertyMetadata->withReadableLink(null !== $normalizationGroups && !empty(array_intersect($normalizationGroups, $relatedGroups)));
         }
 
+        $inputClass = $this->getInputClass($relatedClass) ?? $relatedClass;
+        $relatedGroups = $this->getClassSerializerGroups($inputClass);
```

Comment:
are `$outputClass` and `$inputClass` used? Not sure we need a variable for this?

Reply:
I inlined the variables.